cpp/modmesh/inout/inout_util.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <regex>
+#include <string>
+#include <sstream>
+#include <vector>
+
+#include <modmesh/buffer/buffer.hpp>
+
+namespace modmesh
+{
+
+namespace inout
+{
+
+std::vector<std::string> tokenize(const std::string & str, const std::regex & regex_delim);
+std::vector<std::string> tokenize(const std::string & str, const char delim);
```

Comment:
I have changed the return type to `std::vector<std::string>`.

Reply:
This is nice.

In the future, we can further enhance to use a custom string I/O facility.  String I/O is usually a performance hotspot.  A naive implementation using `std::vector` and `std::string` would not be fast.  But we should not do it without proper profiling.  Let's just keep it in mind, but refrain from doing anything now.